streams/src/test/java/org/apache/kafka/test/KStreamTestDriver.java

```diff
@@ -189,7 +190,23 @@ private ProcessorNode sourceNodeByTopicName(final String topicName) {
         return topicNode;
     }
 
-    public void punctuate(final long timestamp) {
+    public void punctuate(final long timestamp, Punctuator punctuator) {
```

Comment:
This is going to call punctuate on every node - right? Which is not actually correct, I think something like @guozhangwang suggested might be a better approach.